[2024-12-24T17:16:32.627+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-12-24T17:16:32.669+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: streaming_pipeline_only_data.cdc_spark_streaming_to_datalake manual__2024-12-24T17:16:28.417748+00:00 [queued]>
[2024-12-24T17:16:32.681+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: streaming_pipeline_only_data.cdc_spark_streaming_to_datalake manual__2024-12-24T17:16:28.417748+00:00 [queued]>
[2024-12-24T17:16:32.682+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-12-24T17:16:32.698+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): cdc_spark_streaming_to_datalake> on 2024-12-24 17:16:28.417748+00:00
[2024-12-24T17:16:32.708+0000] {standard_task_runner.py:63} INFO - Started process 14718 to run task
[2024-12-24T17:16:32.711+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'streaming_pipeline_only_data', 'cdc_spark_streaming_to_datalake', 'manual__2024-12-24T17:16:28.417748+00:00', '--job-id', '289', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_stream.py', '--cfg-path', '/tmp/tmpvhbkx_r0']
[2024-12-24T17:16:32.713+0000] {standard_task_runner.py:91} INFO - Job 289: Subtask cdc_spark_streaming_to_datalake
[2024-12-24T17:16:32.776+0000] {task_command.py:426} INFO - Running <TaskInstance: streaming_pipeline_only_data.cdc_spark_streaming_to_datalake manual__2024-12-24T17:16:28.417748+00:00 [running]> on host 0b4a5bf62610
[2024-12-24T17:16:32.870+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='streaming_pipeline_only_data' AIRFLOW_CTX_TASK_ID='cdc_spark_streaming_to_datalake' AIRFLOW_CTX_EXECUTION_DATE='2024-12-24T17:16:28.417748+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-24T17:16:28.417748+00:00'
[2024-12-24T17:16:32.871+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-12-24T17:16:32.887+0000] {logging_mixin.py:188} INFO - Checking JAR files...:  True
[2024-12-24T17:16:45.199+0000] {spark_streaming_to_dl.py:123} INFO - Spark session successfully created!
[2024-12-24T17:16:45.216+0000] {spark_streaming_to_dl.py:157} INFO - MinIO configuration is created successfully
[2024-12-24T17:16:47.682+0000] {spark_streaming_to_dl.py:176} INFO - Initial dataframe created successfully!
[2024-12-24T17:16:48.254+0000] {logging_mixin.py:188} INFO - root
 |-- vendorid: integer (nullable = true)
 |-- lpep_pickup_datetime: timestamp (nullable = true)
 |-- lpep_dropoff_datetime: timestamp (nullable = true)
 |-- store_and_fwd_flag: string (nullable = true)
 |-- ratecodeid: double (nullable = true)
 |-- pulocationid: integer (nullable = true)
 |-- dolocationid: integer (nullable = true)
 |-- passenger_count: double (nullable = true)
 |-- trip_distance: double (nullable = true)
 |-- fare_amount: double (nullable = true)
 |-- extra: double (nullable = true)
 |-- mta_tax: double (nullable = true)
 |-- tip_amount: double (nullable = true)
 |-- tolls_amount: double (nullable = true)
 |-- improvement_surcharge: double (nullable = true)
 |-- total_amount: double (nullable = true)
 |-- payment_type: double (nullable = true)
 |-- trip_type: double (nullable = true)
 |-- congestion_surcharge: double (nullable = true)
 |-- id: string (nullable = true)
 |-- id_customer: long (nullable = true)
 |-- rate: double (nullable = true)
[2024-12-24T17:16:48.255+0000] {spark_streaming_to_dl.py:320} INFO - Final dataframe created successfully!
[2024-12-24T17:16:48.256+0000] {spark_streaming_to_dl.py:328} INFO - Streaming is being started...
[2024-12-24T17:18:39.146+0000] {local_task_job_runner.py:310} WARNING - State of this instance has been externally set to None. Terminating instance.
[2024-12-24T17:18:39.147+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-12-24T17:18:39.150+0000] {process_utils.py:132} INFO - Sending 15 to group 14718. PIDs of all processes in the group: [14720, 14718]
[2024-12-24T17:18:39.151+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 14718
[2024-12-24T17:18:39.153+0000] {taskinstance.py:2611} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-12-24T17:18:39.154+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-12-24T17:18:39.168+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/stream_processing/spark_streaming_to_dl.py", line 355, in run_all
    start_steaming(df_final, args.bucket_name, args.year, args.month, args.day)
  File "/opt/airflow/dags/stream_processing/spark_streaming_to_dl.py", line 339, in start_steaming
    return stream_query.awaitTermination()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/streaming/query.py", line 221, in awaitTermination
    return self._jsq.awaitTermination()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1321, in __call__
    answer = self.gateway_client.send_command(command)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1038, in send_command
    response = connection.send_command(command)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/clientserver.py", line 511, in send_command
    answer = smart_decode(self.stream.readline()[:-1])
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/socket.py", line 708, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2613, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-12-24T17:18:39.180+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=streaming_pipeline_only_data, task_id=cdc_spark_streaming_to_datalake, run_id=manual__2024-12-24T17:16:28.417748+00:00, execution_date=20241224T171628, start_date=20241224T171632, end_date=20241224T171839
[2024-12-24T17:18:39.202+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 289 for task cdc_spark_streaming_to_datalake ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(streaming_pipeline_only_data, cdc_spark_streaming_to_datalake, manual__2024-12-24T17:16:28.417748+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'cdc_spark_streaming_to_datalake', 'dag_id': 'streaming_pipeline_only_data', 'run_id': 'manual__2024-12-24T17:16:28.417748+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 12, 24, 17, 16, 32, 670771, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2024, 12, 24, 17, 18, 39, 180006, tzinfo=Timezone('UTC')), 'duration': 126}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 14718)
[2024-12-24T17:18:39.245+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=14718, status='terminated', exitcode=1, started='17:16:32') (14718) terminated with exit code 1
[2024-12-24T17:18:39.874+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=14720, status='terminated', started='17:16:32') (14720) terminated with exit code None
